app-server/app/src/main/scala/token/Routes.scala

```diff
@@ -0,0 +1,102 @@
+package com.azavea.rf.token
+
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.headers.{Authorization, GenericHttpCredentials}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model.Uri.{Path, Query}
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.unmarshalling.Unmarshal
+
+import spray.json.DefaultJsonProtocol
+
+import com.azavea.rf.auth.Authentication
+import com.azavea.rf.utils.{Config, UserErrorHandler}
+
+
+/**
+  * Routes for tokens
+  */
+trait TokenRoutes extends Authentication
+    with UserErrorHandler
+    with Config
+    with SprayJsonSupport
+    with DefaultJsonProtocol {
+
+  import com.azavea.rf.AkkaSystem._
+
+  case class DeviceCredential(id: String, device_name: String)
+  implicit val deviceCredentialJson = jsonFormat2(DeviceCredential)
+
+  case class AuthorizedToken(id_token: String, expires_in: Int, token_type: String)
+  implicit val authorizedTokenJson = jsonFormat3(AuthorizedToken)
+
+  case class RefreshToken(refresh_token: String)
+  implicit val refreshTokenJson = jsonFormat1(RefreshToken)
+
+  val uri = Uri(s"https://$auth0Domain/api/v2/device-credentials")
+  val headers = List(
+    Authorization(GenericHttpCredentials("Bearer", auth0Bearer))
+  )
+
+  def tokenRoutes: Route = pathPrefix("api" / "tokens") {
+    handleExceptions(userExceptionHandler) {
+      pathEndOrSingleSlash {
+        authenticate { user =>
+          get {
+            val params = Query(
+              "type" -> "refresh_token",
+              "user_id" -> user.id
+            )
+            val req = Http().singleRequest(HttpRequest(GET, uri.withQuery(params), headers))
+
+            onSuccess(req) {
+              case HttpResponse(OK, _, entity, _) => complete(Unmarshal(entity).to[List[DeviceCredential]])
```

Comment:
The interception here is not completely useless, even though its ultimate utility is questionable. The response we get from Auth0 is of the format:

```json
[
    {
        "device_name": "ttuhinanshu-macbook-pro",
        "id": "dcr_EEUkyRHihMspypXG",
        "type": "refresh_token",
        "user_id": "auth0|57f..."
    }
]
```

Of these, `id` and `device_name` are legitimately valuable. The `type` will always be `refresh_token`, and the `user_id` will always be the user's id, thus there is no use sending these to the client. However, while those bits are useless, they are also harmless, and we may be doing extra work to remove them. It is worth considering if this interception is justifiable, and whether we should just pipe the request through.